### PR TITLE
Quote string for special characters. Fixes #3214

### LINKED ISSUE #3214: Deploying Static Files. Could not parse theme static file
Hey guys,

I am trying to deploy static files in my magento website that I just installed (2.0.2), I use the command: 

> php -f bin/magento setup:static-content:deploy en_US

And it gives me the following error:

`Could not parse theme static file 'C:/Program Files (x86)/Zend/Apache2/htdocs/ma
gento202/vendor/magento/theme-frontend-blank/web/css/email-fonts.less'>`

I found the file and it's there... what shall I do?

### PR DESCRIPTION
**Pull request by reillo**
The path might contain regular expression character (i.e `(`)  which might cause UnexpectedValueException "Could not parse theme static file". Refer to #3214

### Description
Fix issue of UnexpectedValueException "Could not parse theme static file" when 'bin/magento setup:static-content:deploy -f' is executed. This will also cause of missing static files.

### Fixed Issues (if relevant)

1. magento/magento2#3214: Deploying Static Files. Could not parse theme static file

### Manual testing scenarios

1. Put the Magento Application instance into the folder: C:/Program Files (x86)
2. Run `bin/magento setup:static-content:deploy -f`

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=13344) <br/>All committers have signed the CLA.

**Comment by miguelbalparda** [comment]
Windows is not supported as an operating system per [DevDocs](http://devdocs.magento.com/guides/v2.2/install-gde/system-requirements-tech.html#operating-systems-linux-x86-64). Do you foresee any other cases where this might be relevant?

**Comment by reillo** [comment]
@miguelbalparda But it does work in Windows. That patch is not for windows only, it applies to all OS system. See the commit.

**Comment by reillo** [comment]
@magento-cicd2 CLA signed.

**Comment by miguelbalparda** [comment]
Your description only mentioned Windows, that's why I'm asking. I'm processing a similar [PR](https://github.com/magento/magento2/pull/13340) and I the same might apply.

File: lib/internal/Magento/Framework/App/Utility/Files.php
Comment (oliverklee) [inline lib/internal/Magento/Framework/App/Utility/Files.php]: You'll also need to provide the second parameter to `preg_quote` to make sure the chose delimiter `#` gets escaped, too (so that the escape is always 100% correct).

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Comment by miguelbalparda** [comment]
Your description only mentioned Windows, that's why I'm asking. I'm processing a similar [PR](https://github.com/magento/magento2/pull/13340) and the same might apply.
To avoid any misunderstandings, while Magento 2 might work in Windows, it's not officially supported and that's why we usually reject PRs containing fixes for unsupported systems. 
I'll check this internally and get back to you. Thanks!

File: lib/internal/Magento/Framework/App/Utility/Files.php
Comment (reillo) [reply lib/internal/Magento/Framework/App/Utility/Files.php]: Yep, right. Thanks @oliverklee

**Comment by reillo** [comment]
@miguelbalparda Thanks, miguel. I'll be more specific next time. I'll update the description.

**Comment by orlangur** [comment]
@reillo,
> But it does work in Windows

Actually, not fully, there are some known issues. It was already discussed in some older issues that for good it would be nice to have a proper OS check similar to the PHP version one. Before such check is implemented Windows-only issues and pull requests can be simply closed with an explanation.

So, before any fix can be actually merged there should be a corresponding _VALID_ bug scenario first.

**Review by oliverklee — APPROVED** [review]

**Comment by miguelbalparda** [comment]
To make things clear, @oliverklee is an independent developer who asked for changes in this PR.

**Comment by oliverklee** [comment]
@miguelbalparda That's correct. :-)

As I'm quite new to the Magento project: Is it okay/helpful if I approve PRs that look okay to me, or should this task be limited to dedicated reviewers, core developers or something like this?

**Comment by miguelbalparda** [comment]
I actually asked the same question today in the maintainers channel :)
Basically, feel free to ask for changes where you see fit and keep in mind the final decision is made by maintainers. Thanks!

**Comment by miguelbalparda** [comment]
I'll close this for now, will revisit in the future if anything changes. Thanks @reillo!

**Comment by kanduvisla** [comment]
Why isn't this fix applied? I'm trying to deploy from a Jenkins server (which runs on Linux), from with a Docker container (that is exactly the proper setup for Magento 2) and I'm also getting this error.

If I understand the issue correctly the issue is that I have parenthesis in my path `/var/lib/jenkins/workspace/prik-en-tik/Job Name (acceptance)/vendor/magento/theme-frontend-blank/web/css/_styles.less`. A simple `preg_quote()` fixes this. Heck, you should **always** `preg_quote()` variable input inside a regex!

I'm running into this issue in Magento 2.1.12 by the way. I vote to re-open this issue.

**Pull request closed without merging**
